File: cpp/s02-le-02-conditions.cpp
/*
Cours S02 - 2. Conditions

Multiplication d'un boolean et affichage
*/
#include <iostream>
using namespace std;

int main()
{
	int a(5);
	int b(2);
	int c(5);
	bool d(true);
	bool e(false);
	
	cout << "(a == b) = " << (a == b) << endl;
	cout << "(a == c) = " << (a == c) << endl;
	// overload for bool<< make implicit cast to int ?
	cout << "d = " << d << " and then (bool)d = " << (bool)d << endl;
	
	// guessing that there is a cast
	cout << "2 * (a == b) = " << 2 * (a == b) << endl;
	cout << "2 * (a == c) = " << 2 * (a == c) << endl;
	
	e = a;
	cout << "e = " << e << endl;
	
	return 0;
}

File: python/s02-le-02-conditions.py
#!/usr/bin/env python3
"""
Cours S02 - 2. Conditions

Multiplication d'un boolean et affichage
"""

a = 5;
b = 2;
c = 5;
d = True;
e = False;

print("(a == b) = " + str(a == b));
print("(a == c) = " + str(a == c));
# overload for bool<< make implicit cast to int ?
print("d = " + str(d) + " and then (bool)d = " + str((bool(d))));

#guessing that there is a cast
print("2 * (a == b) = " + str(2 * (a == b)));
print("2 * (a == c) = ");
print(2 * (a == c));

print(type(a))
print(type(e))
e = a;
print("e = " + str(e))
print(type(e))
